Scripts/DCS-BIOS/lib/modules/aircraft_modules/SA342.lua

```diff
@@ -0,0 +1,556 @@
+SA342:defineFloat("STBY_ART_HOR_FLAG", 211, { 0, 1 }, "Raw Gauge Values", "Standby Artificial Horizon Flag")
+SA342:defineFloat("STBY_ART_HOR_W", 212, { 0, 1 }, "Raw Gauge Values", "Standby Artificial Horizon W")
+SA342:defineFloat("STBY_ART_HOR_CURSEUR", 217, { 0, 1 }, "Raw Gauge Values", "Standby Artificial Horizon Curseur")
+SA342:defineFloat("FUEL", 137, { 0.093, 0.932 }, "Raw Gauge Values", "Fuel")
+SA342:defineFloat("ART_HORIZ_PITCH", 27, { -1, 1 }, "Raw Gauge Values", "Artificial Horizon Pitch")
+SA342:defineFloat("ART_HORIZ_ROLL", 28, { -1, 1 }, "Raw Gauge Values", "Artificial Horizon Roll")
+SA342:defineFloat("ART_HORIZ_BALL", 20, { -1, 1 }, "Raw Gauge Values", "Artificial Horizon Ball")
+SA342:defineFloat("ART_HORIZ_FLAG_GS", 18, { 0, 1 }, "Raw Gauge Values", "Artificial Horizon Flag GS")
+SA342:defineFloat("ART_HORIZ_FLAG_HS", 19, { 0, 1 }, "Raw Gauge Values", "Artificial Horizon Flag HS")
+SA342:defineFloat("ART_HORIZ_FLAG_LOC", 29, { 0, 1 }, "Raw Gauge Values", "Artificial Horizon Flag LOC")
+SA342:defineFloat("ART_HORIZ_CURSEUR", 117, { 0, 1 }, "Raw Gauge Values", "Artificial Horizon Curseur")
+SA342:defineFloat("ART_HORIZ_W", 120, { -1, 1 }, "Raw Gauge Values", "Artificial Horizon W")
+SA342:defineFloat("ART_HORIZ_VERT_BAR", 118, { -1, 1 }, "Raw Gauge Values", "Artificial Horizon Vertical Bar")
+SA342:defineFloat("ART_HORIZ_HORIZONTAL_BAR", 119, { -1, 1 }, "Raw Gauge Values", "Artificial Horizon Horizontal Bar")
+SA342:defineFloat("VARIOMETER_VVI", 24, { -0.481, 0.463 }, "Raw Gauge Values", "Variometer Vertical Velocity VVI")
+SA342:defineFloat("IND_AIR_SPEED_IAS", 51, { 0, 0.886 }, "Raw Gauge Values", "Indicated Air Speed IAS")
+SA342:defineFloat("TURBINE_RPM", 135, { 0.095, 0.909 }, "Raw Gauge Values", "Turbine RPM")
+SA342:defineFloat("ROTOR_RPM", 52, { 0.096, 0.904 }, "Raw Gauge Values", "Rotor RPM")
+SA342:defineFloat("VOLT_METER", 14, { 0.045, 0.724 }, "Raw Gauge Values", "Volt Meter")
+SA342:defineFloat("QUATRE", 15, { 0.1575, 0.8625 }, "Raw Gauge Values", "Quatre")
+SA342:defineFloat("TEMP_EXTERNAL", 25, { -0.758, 0.712 }, "Raw Gauge Values", "External Temperature")
+SA342:defineFloat("TEMP_ENGINE_OIL", 151, { 0.046, 0.715 }, "Raw Gauge Values", "Engine Oil Temperature")
+SA342:defineFloat("AUX_FUEL_TANK", 152, { 0, 1 }, "Raw Gauge Values", "Auxiliary Fuel Tank")
+SA342:defineFloat("AM_RADIO_FREQ_100S", 133, { 0, 1 }, "Raw Gauge Values", "AM Radio Frequency 100s")
+SA342:defineFloat("AM_RADIO_FREQ_10S", 134, { 0, 1 }, "Raw Gauge Values", "AM Radio Frequency 10s")
+SA342:defineFloat("AM_RADIO_FREQ_1S", 136, { 0, 1 }, "Raw Gauge Values", "AM Radio Frequency 1s")
+SA342:defineFloat("AM_RADIO_FREQ_TENTHS", 138, { 0, 1 }, "Raw Gauge Values", "AM Radio Frequency Tenths")
+SA342:defineFloat("AM_RADIO_FREQ_HUNDREDTHS", 139, { 0, 1 }, "Raw Gauge Values", "AM Radio Frequency Hundredths")
+
+--Actual Gauge Values
+local function getHDG()
+	local returnValue = (GetDevice(0):get_argument_value(26)) * 360
```

Comment:
There are a lot of GetDevice(0) calls that can be replaced by adding dev0 as a function parameter.

Reply:
and have a ```local dev0 = GetDevice(0)```?

Reply:
Ah I get it, the ```defineIntegerFromGetter``` will supply the dev0.

Reply:
Yup looks good to me!